test/flag_override_test.exs

```diff
@@ -0,0 +1,188 @@
+defmodule ConfigCat.FlagOverrideTest do
+  use ExUnit.Case, async: true
+
+  import Mox
+
+  alias ConfigCat.{Client, LocalFileDataSource, LocalMapDataSource, MockCachePolicy}
+
+  @moduletag capture_log: true
+
+  @cache_policy_id :cache_policy_id
+
+  setup do
+    config = Jason.decode!(~s(
+      {
+        "p": {"u": "https://cdn-global.configcat.com", "r": 0},
+        "f": {
+          "fakeKey": {"v": false, "t": 0, "p": [],"r": []}
+        }
+      }
+    ))
+
+    MockCachePolicy
+    |> stub(:get, fn @cache_policy_id -> {:ok, config} end)
+
+    {:ok, config: config}
+  end
+
+  describe "local-only mode" do
+    test "uses flag values from a local file" do
+      filename = fixture_file("test.json")
+      overrides = LocalFileDataSource.new(filename, :local_only)
+
+      {:ok, client} = start_client(overrides)
+
+      assert Client.get_value(client, "enabledFeature", false) == true
+      assert Client.get_value(client, "disabledFeature", true) == false
+      assert Client.get_value(client, "intSetting", 0) == 5
+      assert Client.get_value(client, "doubleSetting", 0.0) == 3.14
+      assert Client.get_value(client, "stringSetting", "") == "test"
+    end
+
+    test "uses flag values from a simple-format local file" do
+      filename = fixture_file("test_simple.json")
+      overrides = LocalFileDataSource.new(filename, :local_only)
+
+      {:ok, client} = start_client(overrides)
+
+      assert Client.get_value(client, "enabledFeature", false) == true
+      assert Client.get_value(client, "disabledFeature", true) == false
+      assert Client.get_value(client, "intSetting", 0) == 5
+      assert Client.get_value(client, "doubleSetting", 0.0) == 3.14
+      assert Client.get_value(client, "stringSetting", "") == "test"
+    end
+
+    test "reloads file when modified" do
+      flags = %{"flags" => %{"enabledFeature" => false}}
+      filename = temporary_file("simple")
+      overrides = LocalFileDataSource.new(filename, :local_only)
+
+      {:ok, client} = start_client(overrides)
+
+      File.open(filename, [:write], fn file ->
+        IO.write(file, Jason.encode!(flags))
+      end)
+
+      # Backdate file modification time so that it will be different when we
```

Comment:
I like this idea of avoiding the sleep. I'll propagate this change in the other SDKs 👍 .